okhttp/src/main/java/okhttp3/Handshake.kt

```diff
@@ -72,33 +72,26 @@ data class Handshake private constructor(
 
   override fun toString(): String {
     return "Handshake{" +
-        "tlsVersion=" +
-        tlsVersion +
-        " cipherSuite=" +
-        cipherSuite +
-        " peerCertificates=" +
-        names(peerCertificates) +
-        " localCertificates=" +
-        names(localCertificates) +
-        "}"
+        "tlsVersion=$tlsVersion " +
+        "cipherSuite=$cipherSuite " +
+        "peerCertificates=${peerCertificates.map { it.name }} " +
+        "localCertificates=${localCertificates.map { it.name }}}"
   }
 
-  private fun names(certificates: List<Certificate>): List<String> {
-    return certificates
-        .map {
-          if (it is X509Certificate) {
-            it.subjectDN.toString()
-          } else {
-            it.type
-          }
-        }
-  }
+  private val Certificate.name: String
+    get() {
+      return when (this) {
+        is X509Certificate -> subjectDN.toString()
+        else -> type
+      }
+    }
 
   companion object {
     @Throws(IOException::class)
     @JvmStatic
     fun get(session: SSLSession): Handshake {
-      val cipherSuiteString = session.cipherSuite ?: throw IllegalStateException("cipherSuite == null")
+      val cipherSuiteString =
+          session.cipherSuite ?: throw IllegalStateException("cipherSuite == null")
```

Comment:
nit: `checkNotNull(session.cipherSuite) { "cipherSuite == null" }`

Reply:
Fixed.

Reply:
That checkNotNull is nice. TIL